### ISSUE TITLE
Upgrade to newest 4.1 release fails

### ISSUE THREAD

**Issue by Menrath** [academic-moodle-cooperation/moodle-mod_checkmark#75]
Moodle Version:  '4.1.2+ (Build: 20230406)'
PHP-Version: 8.0

```
-->mod_checkmark
Default exception handler: column "checkmark_overrides-&gt;timecreated" cannot be modified. Dependency found with index "mdl_checover_tim_ix (timecreated)" Debug: 
Error code: ddldependencyerror
* line 734 of /lib/ddl/database_manager.php: ddl_dependency_exception thrown
* line 654 of /lib/ddl/database_manager.php: call to database_manager->check_field_dependencies()
* line 1313 of /mod/checkmark/db/upgrade.php: call to database_manager->change_field_default()
* line 934 of /lib/upgradelib.php: call to xmldb_checkmark_upgrade()
* line 677 of /lib/upgradelib.php: call to upgrade_plugins_modules()
* line 1953 of /lib/upgradelib.php: call to upgrade_plugins()
* line 202 of /admin/cli/upgrade.php: call to upgrade_noncore()

!!! column "checkmark_overrides-&gt;timecreated" cannot be modified. Dependency found with index "mdl_checover_tim_ix (timecreated)" !!!
!! 
Error code: ddldependencyerror !!
!! Stack trace: * line 734 of /lib/ddl/database_manager.php: ddl_dependency_exception thrown
* line 654 of /lib/ddl/database_manager.php: call to database_manager-&gt;check_field_dependencies()
* line 1313 of /mod/checkmark/db/upgrade.php: call to database_manager-&gt;change_field_default()
* line 934 of /lib/upgradelib.php: call to xmldb_checkmark_upgrade()
* line 677 of /lib/upgradelib.php: call to upgrade_plugins_modules()
* line 1953 of /lib/upgradelib.php: call to upgrade_plugins()
* line 202 of /admin/cli/upgrade.php: call to upgrade_noncore()
 !!
```

**Comment by mwreichel** [comment]
Thanks for reporting - we will have a look.

**Comment by lukasnoebauer** [comment]
@MWRAMC hi, any progress on this bug? we're encountering it during the upgrade to moodle 4.1 (php 8, postgresql)

**Comment by mwreichel** [comment]
Hi, please excuse my late reply.
We are currently working on this issue and should have a solution soon.

**Comment by bap-dev** [comment]
Hi, just for info: dropping manually the index mdl_checover_tim_ix from the db did the trick.

**Comment by mwreichel** [comment]
Hi,
sorry for our late answer. We weren't able to fix it earlier for different reasons.
It is fixed now with version 4.1.1.
